crates/bevy_reflect/src/lib.rs

```diff
@@ -1995,6 +1995,38 @@ bevy_reflect::tests::Test {
         );
     }
 
+    #[test]
+    fn assert_impl_reflect_on_all() {
```

Comment:
Could you add `assert_impl_all!` assertions for these types? Specifically to test that each receives their corresponding trait (i.e. `Struct` for the struct, `Enum` for the enum, etc.)? Maybe even to verify that they do indeed get all the other traits, like `Typed` and friends.

I can't imagine we'd ever break something that big haha but could be nice to have a sanity check here.

Reply:
i added `assert_impl_all` for `Reflect`, which is sensible, but I don't see how we'd miss the others without also breaking `#[derive(Reflect)]`.